File: cpp_version/src/motorista.cpp
#include "motorista.hpp"

Motorista::Motorista(int idMotorista, std::string nome, std::string cpf, int idade, int anosTrabalho, Contato* contato, Endereco* endereco, float salario) : Pessoa(nome, cpf, contato, endereco) {
   this->idMotorista = idMotorista;
   this->setMotorista(nome, cpf, idade, anosTrabalho, contato, endereco, salario, true);
};

void Motorista::setMotorista(std::string nome, std::string cpf, int idade, int anosTrabalho, Contato* contato, Endereco* endereco, float salario, bool disponibilidade) {
   this->modPessoa(nome, cpf, contato, endereco);
   this->idade = idade;
   this->anosTrabalho = anosTrabalho;
   this->salario = salario;
   this->disponibilidade = disponibilidade;
};

int Motorista::getId() {
   return this->idMotorista;
};

int Motorista::getIdade() {
   return this->idade;
};

int Motorista::getAnosTrabalhando() {
   return this->anosTrabalho;
};

float Motorista::getSalario(){
   return this->salario;
};

bool Motorista::getDisponibilidade() {
   return this->disponibilidade;
};

void Motorista::setDisponibilidade(bool novaDisponibilidade) {
   this->disponibilidade = novaDisponibilidade;
};

File: python/motorista.py
from contato import *
from endereco import *

class Motorista:
    def __init__ (self, idMoto, cpf, nome, idade, anosTrabalho, contato, endereco, salario):
        self.idMoto = idMoto                    #int
        self.cpf = cpf                          #str
        self.nome = nome                        #str
        self.idade = idade                      #int
        self.anosTrabalho = anosTrabalho        #int
        self.salario = salario                  #float
        self.disponibilidade = True             #boolean
        self.contato = contato                  #Contato
        self.endereco = endereco                #Endereco

    #Modificar a idade
    def modIdade(self, novaIdade):
        self.idade = novaIdade

    #Modificar os anos de trabalho
    def modAnosTrabalho(self, novoAnosTrabalho):
        self.anosTrabalho = novoAnosTrabalho

    #Modificar o salario
    def modSalario(self, novoSalario):
        self.salario = novoSalario

    #Modificar disponibilidade em booleano (True/False)
    def modDisponibilidade(self, novaDisponibilidade):
        self.disponibilidade = novaDisponibilidade

    def infoParciais (self):
        return 'ID: ' + str(self.idMoto) + '\nNome: ' + str(self.nome) + '\nIdade: ' + str(self.idade) + '\nAnos de Trabaho: ' + str(self.anosTrabalho) + '\nTelefone para contato: ' + str(self.contato) + '\n'

    def infoCompleta (self):
        return 'ID: ' + str(self.idMoto) + '\nCPF: ' + str(self.cpf) + '\nNome: ' + str(self.nome) + '\nIdade: ' + str(self.idade) + '\nAnos de Trabaho: ' + str(self.anosTrabalho) + '\nContato: ' + str(self.contato) + '\nEndereco: ' + str(self.endereco) + '\nSalario: ' + str(self.salario)
